Evan-Sara-Jalal/movie-app/src/components/moviePage.js

```diff
@@ -0,0 +1,144 @@
+  const [genres, setGenres] = useState([]);
+  const [trailers, setTrailers] = useState([]);
+  const [cast, setCast] = useState([]);
+
+  useEffect(() => {
+    console.log(location);
+    console.log(history);
+
+    fetchMovie();
+    fetchVideos();
+    fetchCredits();
+  }, []);
+
+  const fetchMovie = () => {
+    fetch(
+      `https://api.themoviedb.org/3/movie/${params.id}?api_key=801a0cb3e217f703f966a9186ba74310&language=en-US`
+    )
+      .then((resp) => resp.json())
+      .then((resp) => setMovie(resp));
+  };
+
+  const fetchVideos = () => {
+    fetch(
+      `https://api.themoviedb.org/3/movie/${params.id}/videos?api_key=801a0cb3e217f703f966a9186ba74310&language=en-US`
+    )
+      .then((resp) => resp.json())
+      .then((resp) => setTrailers(resp.results));
+  };
+
+  const fetchCredits = () => {
+    fetch(
+      `https://api.themoviedb.org/3/movie/${params.id}/credits?api_key=801a0cb3e217f703f966a9186ba74310`
+    )
+      .then((resp) => resp.json())
+      .then((resp) => {
+        setCast(resp.cast);
+      });
+  };
+
+  return (
+    <>
+      <Container className="margin-top">
+        <Row className="margin-top">
+          <Link to="/">
+            {" "}
+            <Button variant="danger">Back</Button>{" "}
+          </Link>
+        </Row>
+        <Row>
+          <Col lg={6} md={5} xs={12}>
+            <Image
+              src={`https://image.tmdb.org/t/p/w500/${movie.poster_path}`}
+              rounded
+              fluid
+            />
+          </Col>
+          <Col lg={6} md={5} xs={12}>
+            <h1>{movie.original_title}</h1>
+
+            <h2> Movie overview</h2>
+            {movie.overview}
+            <hr />
+            <h2> Movie Popularity</h2>
+            {movie.popularity}
+            <hr />
+            <h2> Status</h2>
+            {movie.status}
+            <hr />
+            <h2> Genres</h2>
+            <hr />
+
+            {movie.id &&
+              movie.genres.map((genre) => {
+                return <li key={genre.id}>{genre.name}</li>;
+              })}
+            <hr />
+          </Col>
+        </Row>
+      </Container>
+      <Container>
+        <Row>
+          <Col>
+            <h2> Trailers</h2>
+            <hr />
+            {movie.id &&
+              trailers.map((trailer) => {
+                return (
+                  <div className="embed-responsive embed-responsive-16by9 mb-4">
+                    <iframe
+                      className="embed-responsive-item"
+                      src={`https://www.youtube.com/embed/${trailer.key}`}
+                      allowFullScreen
+                      key={trailer.id}
+                    />
+                  </div>
+                );
+              })}
+          </Col>
+        </Row>
+        <br />
+
+        <br />
+        <Row>
+          <Col className="mb-4">
+            <h2>Actors</h2>
+            <hr />
+            {movie.id &&
+              cast.map((actor) => {
```

Comment:
I will prefer it if you used slice() with map() here, and show more info about the actor.